Sources/Networking/HTTPClient/HTTPRequestPath.swift

```diff
@@ -0,0 +1,187 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  HTTPRequestPath.swift
+//
+//  Created by Nacho Soto on 8/8/23.
+
+import Foundation
+
+protocol HTTPRequestPath {
+
+    /// The base URL for requests to this path.
+    static var serverHostURL: URL { get }
+
+    /// Whether requests to this path are authenticated.
+    var authenticated: Bool { get }
+
+    /// Whether requests to this path can be cached using `ETagManager`.
+    var shouldSendEtag: Bool { get }
+
+    /// Whether the endpoint will perform signature verification.
+    var supportsSignatureVerification: Bool { get }
+
+    /// Whether endpoint requires a nonce for signature verification.
+    var needsNonceForSigning: Bool { get }
+
+    /// The relative path component for this endpoint.
+    var relativePath: String { get }
+
+}
+
+extension HTTPRequestPath {
+
+    func url(proxyURL: URL?) -> URL {
+        let baseURL = proxyURL ?? Self.serverHostURL
+        return baseURL.appendingPathComponent(self.relativePath)
+    }
+
+}
+
+// MARK: - Main paths
+
+extension HTTPRequest {
+
+    enum Path: Hashable {
```

Comment:
It is now trivial to add a second `enum` for the new endpoints, which can be passed to `HTTPClient` and it'll make a request to the right place.
That way it will be impossible to make requests to the new server for paths that shouldn't go there.